clients/ui/frontend/src/app/hooks/modelCatalog/useModelCatalogSources.ts

```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import { ModelCatalogContext } from '~/app/context/modelCatalog/ModelCatalogContext';
+import { ModelCatalogContextType } from '~/app/modelCatalogTypes';
+
+export const useModelCatalogSources = (): ModelCatalogContextType => {
+  const context = React.useContext(ModelCatalogContext);
+
+  return context;
```

Comment:
This may be unneccessary if all it does is return the whole context, unless we expect to add more to context later and exclude it from this hook.

Reply:
Same thing as I commented above, you guys should start doing the suggested pattern for development:
API Design -> STUB endpoints in BFF -> frontend + BFF implementation

That way you don't need to have these weird hooks and components half baked, we used to do that in main dashboard due to having all business logic in the frontend but this is not needed anymore.

Given said that, it's fine, although I would suggest adding TODOs in all of those tasks.